classes/Language.php

```diff
@@ -1262,6 +1263,30 @@ public static function installLanguagePack($iso, $params, &$errors = [])
         return count($errors) ? $errors : true;
     }
 
+    private static function updateCurrenciesCldr(Language $language)
+    {
+        /** @var Currency[] $currencies */
+        $currencies = Currency::getCurrencies(true, false, false);
+        $container = SymfonyContainer::getInstance();
+        /** @var LocaleRepository $localeRepoCLDR */
+        $localeRepoCLDR = $container->get('prestashop.core.localization.cldr.locale_repository');
+
+        foreach ($currencies as $currency) {
```

Comment:
You are recoding a part of what is present in `Currency::refreshLocalizedCurrencyData`, the problem is that the code is duplicated/needs to be modified at several places And you are forgetting some part of (what if the cldr currency is not available  - custom currencies)
You'd better use the existing method with a languages array that will only contain one language (even though it's not optimized it will reduce the risk of splitting the behaviour in case of a feature update)